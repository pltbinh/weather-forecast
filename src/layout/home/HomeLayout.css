.home .content {
  min-height: 420px;
}

.home .header {
  color: white;
  background-image: url('../../asset/header.jpg');
  text-align: center;
}

.home .footer {
  background: #262936;
  color: white;
  font-size: 14px;
  min-height: 50px;
}
