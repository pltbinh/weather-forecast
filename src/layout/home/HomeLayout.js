import React from 'react'
import { Jumbotron } from 'react-bootstrap'

import './HomeLayout.css'
export default function HomeLayout({ children }) {
  return (
    <>
      <header data-testid='homeHeader'>
        <Jumbotron fluid className='header'>
          <h1>Home</h1>
        </Jumbotron>
      </header>
      <div className='content'>
        {children}
      </div>
      <footer data-testid='homeFooter'>
        <div className='footer mb-0'>
          <p>Designed by pltbinh</p>
        </div>
      </footer>
    </>
  )
}