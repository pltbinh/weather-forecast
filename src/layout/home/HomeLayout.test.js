import React from 'react';
import { screen, render } from '@testing-library/react'
import renderer from 'react-test-renderer';

import HomeLayout from './HomeLayout';

describe('FilterComponent', () => {
  it('should render correctly', () => {
    // act
    const component = renderer.create(<HomeLayout />);
    const tree = component.toJSON()

    // assert
    expect(tree).toMatchSnapshot()
  });

  it('should render correct footer and header', () => {
    // act
    render(<HomeLayout />);

    const header = screen.getByTestId('homeHeader')
    const footer = screen.getByTestId('homeFooter')

    // assert
    expect(header).toBeDefined()
    expect(footer).toBeDefined()
  });
})