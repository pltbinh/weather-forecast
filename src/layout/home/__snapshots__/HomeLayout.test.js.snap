// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterComponent should render correctly 1`] = `
Array [
  <header
    data-testid="homeHeader"
  >
    <div
      className="header jumbotron jumbotron-fluid"
    >
      <h1>
        Home
      </h1>
    </div>
  </header>,
  <div
    className="content"
  />,
  <footer
    data-testid="homeFooter"
  >
    <div
      className="footer mb-0"
    >
      <p>
        Designed by pltbinh
      </p>
    </div>
  </footer>,
]
`;
